llvm/lib/CodeGen/MachineCSE.cpp

```diff
@@ -184,7 +184,7 @@ bool MachineCSE::PerformTrivialCopyPropagation(MachineInstr *MI,
       continue;
     bool OnlyOneUse = MRI->hasOneNonDBGUse(Reg);
     MachineInstr *DefMI = MRI->getVRegDef(Reg);
-    if (!DefMI->isCopy())
+    if (!DefMI || !DefMI->isCopy())
```

Comment:
I think it would be better if we moved towards making undef operands a verifier error for SSA MIR. These undefined reg checks really should be unnecessary